wild/src/main.rs

```diff
@@ -1,4 +1,118 @@
+use anyhow::anyhow;
+use anyhow::Context;
+use libc::fork;
+use libc::pid_t;
+use std::env::args;
+use std::ffi::c_int;
+use std::ffi::c_void;
+use std::io::Error;
+use std::process;
+
+/// Run the linker with the supplied arguments.
+///
+/// This implements an optimization to remove the time taken for some shutdown tasks from the time
+/// taken for *this* process to complete.
+///
+/// This is done by forking a sub-process which runs the linker and waiting for communication
+/// back from the sub-process (via a pipe) when the main link task is done (the output file has
+/// been written, but some shutdown tasks remain).
 fn main() -> wild_lib::error::Result {
-    let linker = wild_lib::Linker::from_env()?;
-    linker.run()
+    // skip the program name
+    let mut args: Vec<String> = args().skip(1).collect();
+
+    // The default is to fork a sub-process to do the actual linking
+    let mut fork_subprocess = true;
+
+    // But it can be de-activated using the "--no-fork" option. This option is acted upon here
+    // in main, but is removed and not passed onto the linker for parsing
+    args.retain(|a| {
```

Comment:
Probably the most idiomatic approach for this will be (hopefully soon default feature): https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.extract_if.

Reply:
Didn't know about that. Want to send a minor PR to make more idiomatic?

It's funny how you can look at some rust and suspect "there must be a better way to do this..." Usually happens when it looks like 'C'

Reply:
> Usually happens when it looks like 'C'

Yeah, that's exactly my feeling :)
About `extract_if`, it's an unstable feature so that I would wait for stabilization.